crates/weaver_forge/src/config.rs

```diff
@@ -387,40 +309,446 @@ impl CaseConvention {
     }
 }
 
-impl TargetConfig {
-    pub(crate) fn try_new(loader: &dyn FileLoader) -> Result<TargetConfig, Error> {
-        let weaver_file = loader
-            .load_file(WEAVER_YAML)
-            .map_err(|e| InvalidConfigFile {
-                config_file: WEAVER_YAML.into(),
-                error: e.to_string(),
-            })?;
-        if let Some(weaver_file) = weaver_file {
-            serde_yaml::from_str(&weaver_file).map_err(|e| InvalidConfigFile {
-                config_file: WEAVER_YAML.into(),
-                error: e.to_string(),
-            })
-        } else {
-            Ok(TargetConfig::default())
+impl Default for WeaverConfig {
+    fn default() -> Self {
+        Self {
+            file_name: Default::default(),
+            function_name: Default::default(),
+            arg_name: Default::default(),
+            struct_name: Default::default(),
+            field_name: Default::default(),
+            type_mapping: None,
+            text_maps: None,
+            template_syntax: TemplateSyntax {
+                block_start: Some("{%".to_owned()),
+                block_end: Some("%}".to_owned()),
+                variable_start: Some("{{".to_owned()),
+                variable_end: Some("}}".to_owned()),
+                comment_start: Some("{#".to_owned()),
+                comment_end: Some("#}".to_owned()),
+            },
+            whitespace_control: Default::default(),
+            params: None,
+            templates: None,
+            acronyms: None,
+        }
+    }
+}
+
+impl WeaverConfig {
+    /// Collects all the configurations from the path passed in parameter. The configuration files
+    /// are collected in the order of the enumeration below:
+    /// - the $HOME/.weaver/weaver.yaml file,
+    /// - the <path>/weaver.yaml and its <parent folders>weaver.yaml.
+    pub fn collect_from_path<P: AsRef<Path>>(path: P) -> Vec<FileContent> {
```

Comment:
Should we be exposing this "raw", or is this just pub for testing?

I think you should only be exposing a `pub fn load_config_from_path<P: AsRef<Path>>(path: P) -> Result<WeaverConfig, Error>`

The two functions as separate components seems like exposing implementation details.

Reply:
I agree, this is a strange design. I'm working on improving the interface.

Reply:
Fixed in [add9388](https://github.com/open-telemetry/weaver/pull/231/commits/add9388de8f167e777ac9a07fc6010ec3d7850b0)